### DISCUSSION TITLE
Перенос конфига Windows на linux (Keenetic)

### DISCUSSION THREAD

**Discussion by qhhgmgm** [bol-van/zapret#922] [Q&A]
Необходимо перенести данные фильтры в конфиг Zapret на Keenetic

> --filter-tcp=80 --dpi-desync=fake,multidisorder --dpi-desync-fooling=badseq,md5sig --new ^
--filter-tcp=443 --hostlist="%~dp0list-youtube.txt" --dpi-desync=fake,multidisorder --dpi-desync-split-pos=1,midsld --dpi-desync-repeats=11 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls="%~dp0tls_clienthello_www_google_com.bin" --new ^
--filter-tcp=443 --dpi-desync=fake,multidisorder --dpi-desync-skip-nosni=0 --dpi-desync-split-pos=midsld --dpi-desync-fooling=badseq,md5sig --new ^
--filter-udp=443 --hostlist="%~dp0list-youtube.txt" --dpi-desync=fake --dpi-desync-repeats=11 --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new ^
--filter-udp=443 --dpi-desync=fake --dpi-desync-skip-nosni=0 --new ^
--filter-udp=50000-50099 --ipset="%~dp0ipset-discord.txt" --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-cutoff=n4

Опыта у меня не хватило, Zapret вроде запускается, но всё равно, что на ПК, что на android устройствах ничего не работает. Проблему тут вижу в **hostlist** и **ipset**. Если можете, распишите подробный гайд, что куда и как

**Comment by AnnihilatorGun** [comment]
И как именно добавлять в конфиг аргументы, полученные в blockcheck.sh? Под каждый порт одно и тоже писать?

**Comment by bol-van** [comment]
параметр --ipset под Linux принудительно заблокирован именно из-за таких копипастеров, как вы
необходимо использовать kernel mode ipset-ы через custom script
для дискорда есть готовый custom

так же вижу полное непонимание что там  написано, пути из CMD файлов "%~dp0" под linux не работают

изучайте linux

**Reply by qhhgmgm** [reply]
Я изменял конфиг под linux, и использовал готовый custom script для Discord, однако ничего не работало. Сейчас я проблему уже решил. Оказалось, что я забыл раскомментировать строчку **WS_USER=nobody**. Для тех кому интересно какой конфиг я использовал для linux:

> --filter-tcp=80 --dpi-desync=fake,multidisorder --dpi-desync-fooling=badseq,md5sig --new
--filter-tcp=443 --hostlist=/opt/zapret/ipset/list-youtube.txt --dpi-desync=fake,multidisorder --dpi-desync-split-pos=1,midsld --dpi-desync-repeats=11 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --new
--filter-tcp=443 --dpi-desync=fake,multidisorder --dpi-desync-skip-nosni=0 --dpi-desync-split-pos=midsld --dpi-desync-fooling=badseq,md5sig --new
--filter-udp=443 --hostlist=/opt/zapret/ipset/list-youtube.txt --dpi-desync=fake --dpi-desync-repeats=11 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new
--filter-udp=443 --dpi-desync=fake --dpi-desync-skip-nosni=0

Для Discord: cp **-a /opt/zapret/init.d/sysv/custom.d.examples/50-discord /opt/zapret/init.d/sysv/custom.d/50-discord**
И естественно я перенёс файл с хост листами **list-youtube.txt** в папку **ipset**

**Reply by numbereleven-a** [reply]
> И естественно я перенёс файл с хост листами **list-youtube.txt** в папку **ipset**

лучшн перенести файл вне зоны zapret, иначе после обнолвения программы он удалится

**Reply by bol-van** [reply]
в ipset свои файлы тоже не лучшее решение, потому что при обновлении install_easy их снесет
надо выносить из /opt/zapret
